enumerables.rb

```diff
@@ -0,0 +1,286 @@
+        my_each do |e|
+          if cond == e
+            classflag = true
+            break
+          end
+        end
+      end
+    elsif block_given?
+      my_each do |e|
+        if yield(e)
+          classflag = true
+          break
+        end
+      end
+    elsif cond.nil? and !block_given?
+      classflag = if empty?
+              false
+            else
+              true
+            end
+    else
+      false
+    end
+    classflag
+  end
+
+  # my_none Method
+  def my_none?(cond = nil)
+    classflag = true
+    if !cond.nil?
+      if cond.is_a?(Class)
+        my_each do |e|
+          if e.is_a?(cond)
+            classflag = false
+            break
+          end
+        end
+      elsif cond == Numeric
+        my_each do |e|
+          if e.is_a?(Numeric)
+            classflag = false
+            break
+          end
+        end
+      elsif cond.is_a?(Boolean)
+        my_each do |e|
+          if e == cond
+            classflag = false
+            break
+          end
+        end
+      elsif cond != Class and cond.is_a?(Regexp)
+        my_each do |e|
+          if e.is_a?(String)
+            if e.match?(cond)
+              classflag = false
+              break
+            end
+          end
+        end
+      elsif cond.is_a?(Numeric) || cond.is_a?(String)
+        my_each do |e|
+          if cond == e
+            classflag = false
+            break
+          end
+        end
+      end
+
+    elsif block_given?
+      my_each do |e|
+        if yield(e)
+          classflag = false
+          break
+        end
+      end
+    elsif cond.nil? and !block_given?
+      classflag = if to_a.empty?
+        true
+      else
+        false
+      end
+    else
+      true
+    end
+    classflag
+  end
+
+  # my_none Method (another approach), didn't work because we couldn't use yield on a yield...
+  # def my_none_2?(cond = nil)
+  #   res = false
+  #   if !cond.nil?
+  #     inv = self.my_any?(cond)
+  #   elsif block_given?
+  #     block = yield
+  #     inv = self.my_any? {yield.to_s}
+  #   end
+  #   inv == true ? res = false : res = true
+  #   res
+  #   block
+  # end
+
+  # my_count Method
+  def my_count(cond = nil)
+    counter = 0
+    if block_given?
+      my_each do |f|
+        counter += if yield f
+        end
+        counter
+      end
+    elsif cond.nil?
+      length
+    elsif !cond.nil?
+      my_each do |e|
+        counter += 1 if cond == e
+      end
+      counter
+    end
+  end
+
+  # my_map method
+  def my_map(pro = nil)
+    new_array = []
+    if pro.is_a?(Proc)
+      my_each do |e|
+        new_array.push(pro.call(e))
+      end
+      new_array
+    elsif block_given?
+      my_each do |e|
+        new_array.push(yield e)
+      end
+      new_array
+    else
+      new_array = self
+    end
+    new_array
+  end
+
+  # my_inject Method
+  def my_inject(init = 0)
+    res = init
+    my_each do |e|
+      res = yield res, e
+    end
+    res
+  end
```

Comment:
- [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))

####  my_inject
> Remember to make it work with ranges

- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.
```ruby
(1..5).my_inject(:+) == (1..5).inject(:+) => 15
```
- my_inject is retrieving a LocalJumpError
![image](https://user-images.githubusercontent.com/52209067/95551079-4d013a00-09c7-11eb-9fa5-595cf8d92b48.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/95551032-31962f00-09c7-11eb-8dc9-4e5fdb0e16b5.png)

- [ ] when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method::range

```ruby
(1..5).my_inject(2, :+) == (1..5).inject(2, :+)
```
- my_inject retrieves an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/95551358-e0d30600-09c7-11eb-9099-4db325130c98.png)
- Real inject
![image](https://user-images.githubusercontent.com/52209067/95551420-fcd6a780-09c7-11eb-986b-8440ad0c9b40.png)